File: Hashmap/Two Sum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> d;
        vector<int> result;
        for (int index = 0; index < nums.size(); ++index) {
            if (d.find(target - nums[index]) != d.end()) {
                result.push_back(d[target - nums[index]]);
                result.push_back(index);
                break;
            }
            d[nums[index]] = index;
        }
        return result;
    }
};

File: Hashmap/Two Sum/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> d = new Dictionary<int, int>();
        int[] result = new int[2];
        for (int index = 0; index < nums.Length; ++index) {
            if (d.ContainsKey(target - nums[index])) {
                result[0] = d[target - nums[index]];
                result[1] = index;
                break;
            }
            d[nums[index]] = index;
        }
        return result;
    }
}